exchange/non_bid_reason.go

```diff
@@ -1,29 +1,55 @@
 package exchange
 
+import (
+	"errors"
+	"strings"
+	"syscall"
+
+	"github.com/prebid/openrtb/v20/openrtb3"
+	"github.com/prebid/prebid-server/v2/errortypes"
+)
+
 // SeatNonBid list the reasons why bid was not resulted in positive bid
 // reason could be either No bid, Error, Request rejection or Response rejection
-// Reference:  https://github.com/InteractiveAdvertisingBureau/openrtb/blob/master/extensions/community_extensions/seat-non-bid.md
-type NonBidReason int
+// Reference:  https://github.com/InteractiveAdvertisingBureau/openrtb/blob/master/extensions/community_extensions/seat-non-bid.md#list-non-bid-status-codes
 
 const (
-	NoBidUnknownError                      NonBidReason = 0 // No Bid - General
-	ResponseRejectedGeneral                NonBidReason = 300
-	ResponseRejectedBelowFloor             NonBidReason = 301 // Response Rejected - Below Floor
-	ResponseRejectedCategoryMappingInvalid NonBidReason = 303 // Response Rejected - Category Mapping Invalid
-	ResponseRejectedBelowDealFloor         NonBidReason = 304 // Response Rejected - Bid was Below Deal Floor
-	ResponseRejectedCreativeSizeNotAllowed NonBidReason = 351 // Response Rejected - Invalid Creative (Size Not Allowed)
-	ResponseRejectedCreativeNotSecure      NonBidReason = 352 // Response Rejected - Invalid Creative (Not Secure)
+	ResponseRejectedGeneral                openrtb3.NoBidReason = 300
+	ResponseRejectedBelowFloor             openrtb3.NoBidReason = 301 // Response Rejected - Below Floor
+	ResponseRejectedCategoryMappingInvalid openrtb3.NoBidReason = 303 // Response Rejected - Category Mapping Invalid
+	ResponseRejectedBelowDealFloor         openrtb3.NoBidReason = 304 // Response Rejected - Bid was Below Deal Floor
+	ResponseRejectedCreativeSizeNotAllowed openrtb3.NoBidReason = 351 // Response Rejected - Invalid Creative (Size Not Allowed)
+	ResponseRejectedCreativeNotSecure      openrtb3.NoBidReason = 352 // Response Rejected - Invalid Creative (Not Secure)
+	ErrorTimeout                           openrtb3.NoBidReason = 101 // Error - Timeout
+	ErrorGeneral                           openrtb3.NoBidReason = 100 // Error - General
+	ErrorBidderUnreachable                 openrtb3.NoBidReason = 103 // Error - Bidder Unreachable
 )
 
-// Ptr returns pointer to own value.
-func (n NonBidReason) Ptr() *NonBidReason {
-	return &n
+func errorToNonBidReason(errorCode int) openrtb3.NoBidReason {
```

Comment:
This is only used once for far and is partnered with reading the error type code. Consider merging those two operations into a single function:

```
func errorToNonBidReason(err error) NonBidReason {
  switch errortypes.ReadCode(err) {
  case errortypes.TimeoutErrorCode:
    return ErrorTimeout
  default:
    return ErrorGeneral
  }
}
```

Reply:
Done